src/reporter/timex.py

```diff
@@ -84,23 +65,27 @@ def select_time_index_value(headers: dict, notification: dict) -> datetime:
     :param notification: the notification JSON payload as received from Orion.
     :return: the value to be used as time index.
     """
-    custom_index = to_datetime(
-                            _custom_time_index_attribute(headers, notification))
-    time_instant = to_datetime(_time_instant_attribute(notification))
-    meta_time_instant = latest_from_str_rep(
-                                _iter_time_instant_in_metadata(notification))
-    date_modified = to_datetime(_date_modified_attribute(notification))
-    meta_date_modified = latest_from_str_rep(
-                                _iter_date_modified_in_metadata(notification))
-    default_value = datetime.now()
-
-    priority_list = [
-        custom_index, time_instant, meta_time_instant, date_modified,
-        meta_date_modified, default_value
+    considered_attributes = [
+        maybe_value(headers, TIME_INDEX_HEADER_NAME),
+        "TimeInstant",
+        "timestamp",
+        "dateModified"
     ]
 
-    return [d for d in priority_list if d][0]
-    # Note. Index will never be out of bounds since we added a default value.
+    for attr_name in considered_attributes:
+        if not attr_name:
+            continue
+
+        attr_value = to_datetime(_attribute(notification, attr_name))
+        if attr_value:
+            return attr_value
+
+        meta_value = latest_from_str_rep(
+            _iter_metadata(notification, attr_name))
+        if meta_value:
+            return meta_value
+
+    return datetime.now()
```

Comment:
From chicco's comment, this `now()` call might be better done at the beginning of the whole method.

Reply:
why? sorry I missed that too :-)

Reply:
Yes, to be more accurate the current time should really be saved at the beginning, since processing the time index calculation might take a little time. Should be only a couple of milliseconds at most but still a good point.